### PR TITLE
add deprecation warnings not use some configurations anymore

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
Deprecation warnings for the removal of configuration in #453 
1.2.8 need to be released then
1.x need to merged into master then

### CODE REVIEW

File: DependencyInjection/SonataDoctrinePHPCRAdminExtension.php [hunk 1]
```diff
@@ -89,6 +89,16 @@ public function getNamespace()
...
     {
+        @trigger_error(
+            'The "sonata_doctrine_phpcr_admin.document_tree_options" configuration is deprecated since 1.2.8, to be replaced in 2.0.',
+            E_USER_DEPRECATED
+        );
+
+        @trigger_error(
+            'The "sonata_doctrine_phpcr_admin.document_tree_defaults" configuration is deprecated since 1.2.8, to be moved in 2.0.',
+            E_USER_DEPRECATED
+        );
```
Comment (greg0ire) [inline DependencyInjection/SonataDoctrinePHPCRAdminExtension.php]: Shouldn't there be some kind of condition here?

File: DependencyInjection/SonataDoctrinePHPCRAdminExtension.php [hunk 2]
```diff
@@ -89,6 +89,18 @@ public function getNamespace()
...
+            'The "sonata_doctrine_phpcr_admin.document_tree_options" configuration is deprecated since 1.2.8, to be replaced in 2.0.',
+            E_USER_DEPRECATED
+        );
+
+        if (isset($config['document_tree_defaults'])) {
+            @trigger_error(
+                'The "sonata_doctrine_phpcr_admin.document_tree_defaults" configuration is deprecated since 1.2.8, to be moved in 2.0.',
+                E_USER_DEPRECATED
+            );
+        }
```
Comment (greg0ire) [inline DependencyInjection/SonataDoctrinePHPCRAdminExtension.php]: What should be used instead of the deprecated configuration nodes?

Comment (greg0ire) [inline DependencyInjection/SonataDoctrinePHPCRAdminExtension.php]: This PR is supposed to introduce the thing you are deprecating this in favor of if it is not already available. The PR on master should only do removals.

File: DependencyInjection/SonataDoctrinePHPCRAdminExtension.php [hunk 3]
```diff
@@ -89,6 +89,20 @@ public function getNamespace()
...
+        @trigger_error(
+            'The "sonata_doctrine_phpcr_admin.document_tree_options" configuration is deprecated since 1.2.8, to be replaced in 2.0.',
+            E_USER_DEPRECATED
+        );
+
+        if (isset($config['document_tree_defaults'])) {
+            @trigger_error(
+                'The "sonata_doctrine_phpcr_admin.document_tree_defaults" configuration is deprecated
+                since 1.2.8, to be moved in 2.0.
+                Use "sonata_doctrine_phpcr_admin.document_tree.routing_defaults" then',
```
Comment (greg0ire) [inline DependencyInjection/SonataDoctrinePHPCRAdminExtension.php]: You should introduce `sonata_doctrine_phpcr_admin.document_tree.routing_defaults` in this PR and remove " then". Same goes for the other deprecations

**Comment by dbu** [comment]
you can't deprecate something without offering the new solution. as far as i can tell, we can't do forward compatibility from 1.x to 2.0, so there is no point in my opinion to have any deprecations for this. as long as the upgrade instructions are clean, thats good enough for me.

**Comment by greg0ire** [comment]
Thanks for clearing that up @dbu !

**Pull request closed without merging**